creators/Fenix/StopsCreatorFenix.py

```diff
@@ -1,19 +1,142 @@
 # coding=utf-8
 
+import overpy
+import sys
+
+from transitfeed import util
 from creators.StopsCreator import StopsCreator
+from osmhelper.osm_routes import Route, RouteMaster
+from osmhelper.osm_stops import Stop
 
 
 class StopsCreatorFenix(StopsCreator):
 
     def add_stops_to_schedule(self, schedule, data):
 
-        stops = data.stops
+        # Get stops information
+        stops = data.get_stops()
 
         # add all stops to GTFS
         for stop in stops.values():
+
+            # If there is no name, query one intelligently from OSM
+            if stop.name == Stop.NO_NAME:
+                self._get_interim_stop_name(stop)
+                print stop
+
+            # Add stop to GTFS object
             schedule.AddStop(
                 lat=float(stop.lat),
                 lng=float(stop.lon),
                 name=stop.name,
                 stop_id=str(stop.id)
             )
+
+        # Add loose stop objects to route objects
+        self.add_stops_to_routes(schedule, data)
+
+    def add_stops_to_routes(self, schedule, data):
+
+        routes = data.routes
+        stops = data.stops
+
+        # Loop through routes
+        for ref, route in routes.iteritems():
+            # Replace stop ids with Stop objects
+            self._fill_stops(stops, route)
+
+        data.routes = routes
+        return
+
+    def _fill_stops(self, stops, route):
+        """Fill a route object with stop objects for of linked stop ids
+
+        """
+        if isinstance(route, Route):
+            i = 0
+            for stop in route.stops:
+                if stop in stops:
+                    # Replace stop id with Stop objects
+                    route.stops[i] = stops[stop]
```

Comment:
Yes, I'm with you. There is a basic new concept here. Let me explain:

Before: A route had it's stops as objects assigned. One identical stop, which has been used by different routes was instantiated several times and was handled as different ones. I didn't like the idea. 

Now: With the query optimizations this is not happening anymore. Routes and Stops are handled separately and efficiently. The route objects usually have a dictionary of (only) stop ids to point to the stop objects.

But Fenix current logic expects them to be objects. I would suggest to change Fenix' logic in the future. This replacement from ids to objects is just a temporary way to simulate the conversion which actually should be improved (in my humble opinion) in Fenix' logic.